--- busybox-dist/Makefile	2011-05-12 18:29:59.083554001 -0700
+++ busybox/Makefile	2011-05-12 18:30:26.047553999 -0700
@@ -213,7 +213,6 @@
 
 busybox: .depend $(libraries-y)
 	$(CC) $(EXTRA_CFLAGS) $(LDFLAGS) -o $@ -Wl,--start-group $(libraries-y) $(LIBRARIES) -Wl,--end-group
-	$(STRIPCMD) $@
 
 busybox.links: $(top_srcdir)/applets/busybox.mkll include/bb_config.h $(top_srcdir)/include/applets.h
 	- $(SHELL) $^ >$@
