--- busybox.orig/networking/ping6.c	2008-09-27 18:17:45.000000000 -0700
+++ busybox-1.1.0/networking/ping6.c	2008-09-27 18:18:26.000000000 -0700
@@ -243,9 +243,9 @@ static char *icmp6_type_name (int id)
 	case ICMP6_PARAM_PROB:				return "Parameter Problem";
 	case ICMP6_ECHO_REPLY:				return "Echo Reply";
 	case ICMP6_ECHO_REQUEST:			return "Echo Request";
-	case ICMP6_MEMBERSHIP_QUERY:		return "Membership Query";
-	case ICMP6_MEMBERSHIP_REPORT:		return "Membership Report";
-	case ICMP6_MEMBERSHIP_REDUCTION:	return "Membership Reduction";
+	case MLD_LISTENER_QUERY:		return "Membership Query";
+	case MLD_LISTENER_REPORT:		return "Membership Report";
+	case MLD_LISTENER_REDUCTION:	return "Membership Reduction";
 	default: 							return "unknown ICMP type";
 	}
 }
