--- busybox-1.1.0-old/editors/patch.c	2011-07-13 17:41:32.000000000 -0700
+++ busybox-1.1.0/editors/patch.c	2011-07-14 16:54:16.000000000 -0700
@@ -130,11 +130,22 @@
 			patch_line = bb_get_line_from_file(patch_file);
 		}
 
+		if (!patch_line) {
+			ret = 2;
+			bb_error_msg("Invalid patch");
+			continue;
+		}
+
 		/* Extract the filename used before the patch was generated */
 		original_filename = extract_filename(patch_line, patch_level);
 		free(patch_line);
 
 		patch_line = bb_get_line_from_file(patch_file);
+		if (!patch_line) {
+			ret = 2;
+			bb_error_msg("Invalid patch");
+			continue;
+		}
 		if (strncmp(patch_line, "+++ ", 4) != 0) {
 			ret = 2;
 			bb_error_msg("Invalid patch");
@@ -182,20 +193,27 @@
 		while (patch_line) {
 			unsigned int count;
 			unsigned int src_beg_line;
-			unsigned int unused;
+			unsigned int src_chunk_size = 1;
+			unsigned int dest_chunk_size = 1;
 			unsigned int hunk_offset_start = 0;
 			int hunk_error = 0;
 
 			/* This bit should be improved */
-			if ((sscanf(patch_line, "@@ -%d,%d +%d,%d @@", &src_beg_line, &unused, &dest_beg_line, &unused) != 4) &&
-				(sscanf(patch_line, "@@ -%d,%d +%d @@", &src_beg_line, &unused, &dest_beg_line) != 3) &&
-				(sscanf(patch_line, "@@ -%d +%d,%d @@", &src_beg_line, &dest_beg_line, &unused) != 3)) {
+			if ((sscanf(patch_line, "@@ -%d,%d +%d,%d @@", &src_beg_line, &src_chunk_size, &dest_beg_line, &dest_chunk_size) != 4) &&
+				(sscanf(patch_line, "@@ -%d,%d +%d @@", &src_beg_line, &src_chunk_size, &dest_beg_line) != 3) &&
+				(sscanf(patch_line, "@@ -%d +%d,%d @@", &src_beg_line, &dest_beg_line, &dest_chunk_size) != 3) &&
+				(sscanf(patch_line, "@@ -%d +%d @@", &src_beg_line, &dest_beg_line) != 2)) {
 				/* No more hunks for this file */
 				break;
 			}
 			free(patch_line);
 			hunk_count++;
 
+			if (src_chunk_size == 0)
+				src_beg_line++;
+			if (dest_chunk_size == 0)
+				dest_beg_line++;
+
 			if (src_beg_line && dest_beg_line) {
 				/* Copy unmodified lines upto start of hunk */
 				/* src_beg_line will be 0 if its a new file */
@@ -247,7 +265,7 @@
 
 		/* Cleanup last patched file */
 		if (copy_trailing_lines_flag) {
-			copy_lines(src_stream, dst_stream, -1);
+			dest_cur_line += copy_lines(src_stream, dst_stream, -1);
 		}
 		if (src_stream) {
 			fclose(src_stream);
