commit b23f3f8996db1c12b1c41c02e98010caa4392906
Author: Stephen Segal <ssegal@meraki.com>
Date:   Thu May 26 15:43:22 2016 -0700

    Revert of mainline 9faa7e94b824e1dc5fd9ce1b6f16c1b3f13c601a

diff --git a/libc/stdlib/Makefile.in b/libc/stdlib/Makefile.in
index ae74995..2b06c66 100644
--- a/libc/stdlib/Makefile.in
+++ b/libc/stdlib/Makefile.in
@@ -61,6 +61,7 @@ CSRC-$(if $(findstring yyy,$(UCLIBC_HAS_FLOATS)$(UCLIBC_HAS_WCHAR)$(UCLIBC_HAS_X
 
 # multi source _atexit.c
 CSRC-y += __cxa_atexit.c __cxa_finalize.c __exit_handler.c exit.c on_exit.c
+CSRC-$(COMPAT_ATEXIT) += old_atexit.c
 
 STDLIB_DIR := $(top_srcdir)libc/stdlib
 STDLIB_OUT := $(top_builddir)libc/stdlib
@@ -70,12 +71,10 @@ STDLIB_OBJ := $(patsubst %.c,$(STDLIB_OUT)/%.o,$(CSRC-y))
 
 libc-y += $(STDLIB_OBJ)
 libc-static-y += $(STDLIB_OUT)/atexit.o $(STDLIB_OUT)/system.o
-libc-static-$(COMPAT_ATEXIT) += $(STDLIB_OUT)/old_atexit.o
 libc-shared-y += $(STDLIB_OUT)/system.oS
 
 # this should always be the PIC version, because it could be used in shared libs
 libc-nonshared-y += $(STDLIB_OUT)/atexit.os
-libc-nonshared-$(COMPAT_ATEXIT) += $(STDLIB_OUT)/old_atexit.os
 libc-nomulti-y += $(STDLIB_OUT)/labs.o $(STDLIB_OUT)/atol.o $(STDLIB_OUT)/_stdlib_strto_l.o $(STDLIB_OUT)/_stdlib_strto_ll.o
 libc-nomulti-$(UCLIBC_HAS_XLOCALE) += $(STDLIB_OUT)/_stdlib_strto_l_l.o $(STDLIB_OUT)/_stdlib_strto_ll_l.o
